docs/content/guides/dagster/local_to_production_tutorial.mdx

```diff
@@ -0,0 +1,395 @@
+                "by": "user1",
+            },
+            2: {
+                "id": 2,
+                "type": "story",
+                "title": "an awesome story",
+                "by": "user2",
+            },
+        }
+
+    @property
+    def item_field_names(self):
+        return ["id", "type", "title", "by"]
+
+    def fetch_item_by_id(self, item_id: int) -> Optional[Dict[str, Any]]:
+        return self.data.get(item_id)
+
+    def fetch_max_item_id(self) -> int:
+        return len(self.data.items())
+
+
+@resource
+def mock_hn_client():
+    return MockHNClient()
+```
+
+Now, let’s write tests to verify that the assets produce the datasets we expect:
+
+```python file=/guides/dagster/local_to_prod_tutorial/test_assets.py startafter=start endbefore=end
+# test_assets.py
+
+def test_items():
+    context = build_op_context(resources={"hn_client": mock_hn_client})
+    hn_dataset = items(context)
+    assert isinstance(hn_dataset, pd.DataFrame)
+
+    expected_data = pd.DataFrame(MockHNClient().data.values()).rename(columns={"by": "user_id"})
+
+    assert (hn_dataset == expected_data).all().all()
+
+
+def test_comments():
+    mock_data = pd.DataFrame({"id": [1, 2, 3], "type": ["comment", "story", "comment"]})
+    out_data = comments(mock_data)
+
+    out_ids = out_data["id"].tolist()
+
+    assert 1 in out_ids
+    assert 2 not in out_ids
+    assert 3 in out_ids
+
+
+def test_stories():
+    mock_data = pd.DataFrame({"id": [1, 2, 3], "type": ["story", "story", "comment"]})
+    out_data = stories(mock_data)
+
+    out_ids = out_data["id"].tolist()
+
+    assert 1 in out_ids
+    assert 1 in out_ids
+    assert 3 not in out_ids
+```
+
+### Part Three: Production
+
+Now that our assets work locally and can be tested, let’s set up for the production environment. We want to store the assets in a production Snowflake database, so we need to update the configuration for the `snowflake_io_manager`:
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v3.py startafter=start endbefore=end
+# repository.py
+
+@repository
+def repo():
+    resource_defs = {
+        "hn_client": hn_api_client,
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": "abc1234.us-east-1",
+                "user": "dev@company.com",
+                "password": "company_super_secret_password",
+                "database": "PRODUCTION",
+                "schema": "HACKER_NEWS"
+            }
+        ),
+    }
+
+    return [*with_resources([items, comments, stories], resource_defs=resource_defs)]
+```
+
+Here we run into an issue. The next time a developer wants to work on the assets, they will have to remember to change the configuration for the `snowflake_io_manager`. This leaves room for a developer to accidentally override the production asset during local development. Instead we can set up the repository to attach resources to assets based on the environment:
```

Comment:
```suggestion
Here we run into an issue. The next time a developer wants to work on the assets, they will have to remember to change the configuration for the `snowflake_io_manager`. This leaves room for a developer to accidentally overwrite the production asset during local development.

Instead, we can set up the repository to attach resources to assets based on the environment:
```

Suggesting `overwrite` because it made more sense to me, but feel free to ignore if it's actually incorrect.

Reply:
overwrite is correct! thanks for catching it